mmengine/visualization/visualizer.py

```diff
@@ -398,26 +415,68 @@ def draw_points(self,
         assert positions.shape[-1] == 2, (
             'The shape of `positions` should be (N, 2), '
             f'but got {positions.shape}')
-        colors = color_val_matplotlib(colors)  # type: ignore
-        self.ax_save.scatter(
-            positions[:, 0], positions[:, 1], c=colors, s=sizes, marker=marker)
+        if self.backend == 'matplotlib':
+            assert isinstance(marker, str) or marker is None, (
+                'The type of `marker` in `matplotlib` should be str, '
+                f'but got {type(marker)}')
+            colors = color_val_matplotlib(colors)  # type: ignore
+            self.ax_save.scatter(
+                positions[:, 0],
+                positions[:, 1],
+                c=colors,
+                s=sizes,
+                marker=marker)
+        else:
+            positions = positions.tolist()
+            if isinstance(colors, str) or isinstance(colors, tuple):
+                colors = [colors] * len(positions)
+            colors = color_val_opencv(colors)  # type:ignore
+            if sizes is not None:
+                sizes = tensor2ndarray(sizes)
+            if marker is None:
+                for i, pos in enumerate(positions):
+                    pos = (int(pos[0]), int(pos[1]))
+                    cv2.circle(
+                        img=self._image,
+                        center=pos,
+                        radius=int(sizes[i]) if sizes is not None else 6,
+                        color=colors[i],
+                        thickness=-1)
+            else:
+                assert isinstance(
+                    marker,
+                    int), ('The type of `marker` in `cv2` should be int, '
+                           f'but got {type(marker)}')
+                if sizes is None:
+                    sizes = [20] * len(positions)
+                for pos, color, size in zip(positions, colors, sizes):
+                    kwargs = {
+                        'img': self._image,
+                        'position': pos,
+                        'color': color,
+                        'markerType': marker,
+                        'markerSize': int(size)
+                    }
+                    cv2.drawMarker(**kwargs)
+
         return self
 
     @master_only
     def draw_texts(
         self,
         texts: Union[str, List[str]],
         positions: Union[np.ndarray, torch.Tensor],
-        font_sizes: Optional[Union[int, List[int]]] = None,
-        colors: Union[str, tuple, List[str], List[tuple]] = 'g',
+        font_sizes: Optional[Union[int, float, List[int], List[float]]] = None,
```

Comment:
`List[Union[int, str]]` indicates that this argument could be a list of integers or strings. For instance, `[1, 2]`, `[1.0, 2.0]`, or `[1.0, "two"]` are all valid examples. On the other hand, `Union[List[int], List[float]]` implies a list that can contain either integers or floats. `[1, 2]` and `[1.0, 2.0]` are both acceptable, while `[1.0, 2]` is not permissible.

In this example, there are numerous `List[xxx, yyy]` annotations. Please ensure that all of these arguments are properly set.

https://github.com/open-mmlab/mmengine/pull/1285/files#diff-ddeb044be5acfc012a996b3ff1566f628e566323c456e8838361c63a90b95b72L703
https://github.com/open-mmlab/mmengine/pull/1285/files#diff-ddeb044be5acfc012a996b3ff1566f628e566323c456e8838361c63a90b95b72L703